kitsune/search/v2/base.py

```diff
@@ -22,26 +22,33 @@ def prepare(cls, instance):
 
             # This will allow child classes to have their own methods in the form of prepare_field
             prepare_method = getattr(obj, "prepare_{}".format(f), None)
-            # Ovewrite the value if there is a specific method implented for the specific field
-            if prepare_method is not None:
-                value = prepare_method(instance)
-            else:
-                value = getattr(instance, f)
-
-            # Assign values to each field. Assign a dictionary to Objects
-            if isinstance(doc_mapping.resolve_field(f), field.Object):
+            value = obj.get_field_value(f, instance, prepare_method)
+
+            # Assign values to each field.
+            field_type = doc_mapping.resolve_field(f)
+            if isinstance(field_type, field.Object) and not (
+                isinstance(value, InnerDoc)
+                or (isinstance(value, list) and isinstance((value or [None])[0], InnerDoc))
+            ):
+                # if the field is an Object but the value isn't an InnerDoc
+                # or a list containing an InnerDoc then we're dealing with locales
                 locale = obj.prepare_locale(instance)
                 if locale and value:
                     obj[f] = {locale: value}
-                # We  populate non-locale object fields here
             else:
                 setattr(obj, f, value)
 
-        obj.meta.id = instance.id
+        obj.meta.id = instance.pk
```

Comment:
I like the attention to detail :P

Reply:
It's needed! `Profile`'s pk is its `user_id` field